diff --git a/crates/gitbutler-git/src/repository.rs b/crates/gitbutler-git/src/repository.rs
index 712141e42..40e056ad0 100644
--- a/crates/gitbutler-git/src/repository.rs
+++ b/crates/gitbutler-git/src/repository.rs
@@ -169,7 +169,7 @@ where
             {
                 #[cfg(unix)]
                 {
-                    format!("'{setsid_path}' ")
+                    format!("{setsid_path} ")
                 }
                 #[cfg(windows)]
                 {
@@ -191,6 +191,8 @@ where
         ),
     );

+    envs.insert("GIT_TRACE".into(), "true".into());
+    envs.insert("PATH".into(), "/app/bin:/usr/bin".into());
     let mut child_process = core::pin::pin! {
         async {
             executor
